component/main.jsonnet

```diff
@@ -66,12 +68,34 @@ local readServices = kube.ClusterRole('appcat:services:read') + {
   ],
 };
 
-{
-  '10_clusterrole_view': xrdBrowseRole,
-  [if isOpenshift then '10_clusterrole_finalizer']: finalizerRole,
-  '10_clusterrole_services_read': readServices,
+// adding namespace for syn-appcat
+local ns = kube.Namespace('syn-appcat');
+// adding missing namespace for slos
+local slos_namespace = kube.Namespace('appcat-slos');
+
+// add secret
+
+local secret = kube.Secret('mailgun-smtp-credentials') {
+  metadata+: {
+    namespace: 'syn-appcat',
+  },
+  stringData: {
+    password: pgParams.mailgun_smtp_password,
+  },
+};
+
+if params.services.vshn.enabled && pgParams.enabled then
```

Comment:
This `if` seems to remove those files clusterrole files when generating the golden images. That will probably break a lot of stuff, please change this if statement so that it only applies to the mailgun secret.

Reply:
This still removes some necessary files it seems.